README.md

```diff
@@ -1,9 +1,243 @@
+import time
+
+import bdai_ros2_wrappers.process as ros_process
+
+import std_msgs.msg
+
+@ros_process.main()
+def entrypoint() -> None:
+    # no need to initialize, it is automatic
+    node = ros_process.node()  # or entrypoint.node
+    assert node is not None
+    pub = node.create_publisher(std_msgs.msg.String, "test", 1)
+
+    def callback() -> None:
+        time.sleep(10)  # you can block in a callback
+        pub.publish(std_msgs.msg.String(data="testing"))
+    executor = ros_process.executor()  # or entrypoint.executor
+    assert executor is not None
+    executor.create_task(callback)
+
+    time.sleep(10)  # you can block in the main thread
+
+    node.get_logger().info("testing")
+    logging.info("testing")  # you can use Python logging
+
+    return  # no need to cleanup or shutdown, it is automatic
+
+if __name__ == "__main__":
+    entrypoint()
+```
+
+This is ideal for quick prototyping and simple scripts.
+
+#### Setting up multi-node processes
+
+You can spin as many ROS 2 nodes as you need, and skip the default process-wide node if unnecessary:
+
+```python
+import logging
+import time
+
+from typing import Any, List
+
+import bdai_ros2_wrappers.process as ros_process
+from bdai_ros2_wrappers.node import Node
+
+def graph(**kwargs: Any) -> List[Node]:
+    # make sure to forward all keyword arguments for proper loading!
+    return [Node("my_node", **kwargs), Node("my_other_node", **kwargs)]
+
+@ros_process.main(prebaked=False)
+def main() -> None:
+    ros_process.load(Node, "_hidden_node")  # or main.load
+    ros_process.spin(graph)  # or main.spin, and it will block!
+
+if __name__ == "__main__":
+    main()
+```
+
+Note the use of ROS 2 node(s) factories, to load (or spin) entire collections at once.
+
+#### Setting up interactive multi-node applications
+
+For interactive applications, you will most likely want to keep automatic spinning in place:
+
+```python
+import logging
+import time
+
+from typing import Any, List
+
+from bdai_ros2_wrappers.futures import wait_for_future
+from bdai_ros2_wrappers.node import Node
+import bdai_ros2_wrappers.process as ros_process
+
+def graph(**kwargs: Any) -> List[Node]:
+    # make sure to forward all keyword arguments for proper loading!
+    return [Node("my_node", **kwargs), Node("my_other_node", **kwargs)]
+
+def work(prompt: str) -> None:
+    def worker() -> None:
+        node = ros_process.node()
+        assert node is not None
+        time.sleep(5.0)  # sleeping is safe!
+        node.get_logger().info(prompt + " done!")
+    executor = ros_process.executor()
+    assert executor is not None
+    future = executor.create_task(worker)  # dispatch worker to executor
+    wait_for_future(future)  # block until worker is done
+
+@ros_process.main()
+def main() -> None:
+    ros_process.load(graph)  # or main.load
+    while True:
+        work(input())
+
+if __name__ == "__main__":
+    main()
+```
+
+#### Setting up command-line single node applications
```

Comment:
Could there be a sentence here about what this is showing and also what `managed` means?